gtfs/__main__.py

```diff
@@ -1,25 +1,61 @@
 #!/usr/bin/env python
 """Command line interface for fetching GTFS."""
-import argparse
-import getpass
+import typer
+from typing_extensions import Annotated
+from chalky import chain
 import logging
 import sys
-
-from prettytable import PrettyTable
-
-from gtfs.utils.FeedSource import FeedSource
-import feed_sources
-# import all the available feed sources
-# pylint: disable=I0011,wildcard-import
-from feed_sources import *
+from prettytable.colortable import ColorTable, Themes
+from halo import Halo
+from .feed_source import FeedSource
+from .feed_sources import __all__ as feed_sources
 
 logging.basicConfig()
 LOG = logging.getLogger()
 LOG.setLevel(logging.INFO)
+app = typer.Typer()
 
-def fetch_all(sources=None):
-    """Fetch from all FeedSources in the feed_sources directory.
-
+def check_predicate(value: str):
+    if value not in ['intersects', 'contains']:
+        raise typer.BadParameter('predicate must be one of intersects or contains')
+    return value
+            
+@app.command()
+def list_feeds(bbox: Annotated[str, typer.Argument(help="pass value as a string separated by commas like this: xmin,ymin,xmax,ymax ")],
+    predicate: Annotated[str, typer.Option("--predicate", "-p", help="intersects or contains", case_sensitive=True, callback=check_predicate)] = 'intersects'):
+    """Filter feeds spatially based on bounding box."""
+    if ',' not in bbox or len(bbox.split(',')) != 4:
+        LOG.critical(chain.bright_red | "Please pass bbox as a string separated by commas like this: xmin,ymin,xmax,ymax")
+        sys.exit(1)
+    for coord in bbox.split(','):
+        try:
+            float(coord)
+        except ValueError:
+            LOG.critical(chain.bright_red | "Please pass only numbers as bbox values!")
+            sys.exit(1)
```

Comment:
I bet you can do that in a callback, see https://typer.tiangolo.com/tutorial/options/callback-and-context/#validate-cli-parameters.